src/components/dock/dock.tsx

```diff
@@ -0,0 +1,177 @@
+import {
+    Component,
+    Event,
+    EventEmitter,
+    h,
+    Host,
+    Prop,
+    State,
+} from '@stencil/core';
+import { DockItem } from './dock.types';
+
+const DEFAULT_MOBILE_BREAKPOINT = 700;
+
+/**
+ * @exampleComponent limel-example-dock-basic
+ * @exampleComponent limel-example-dock-custom-component
+ * @exampleComponent limel-example-dock-mobile
+ * @exampleComponent limel-example-dock-expanded
+ * @exampleComponent limel-example-dock-colors-css
+ * @private
+ */
+@Component({
+    tag: 'limel-dock',
+    shadow: true,
+    styleUrl: 'dock.scss',
+})
+export class Dock {
+    /**
+     * Items that are placed in the dock.
+     */
+    @Prop()
+    public dockItems: DockItem[] = [];
+
+    /**
+     * Items that are placed at the bottom of the dock. (Or at the end in mobile
+     * layout.)
+     */
+    @Prop()
+    public dockFooterItems?: DockItem[] = [];
```

Comment:
this may not be a big deal, but this is not the way we do such a thing in `limel-progress-flow`. In there, we put `isOffProgress` on each item which has to go to the "end group": https://github.com/Lundalogik/lime-elements/blob/ebc7b54ee90f43df483f067cd0241e685df5b276/src/components/progress-flow/examples/progress-flow-off-progress-steps.tsx#L49

So we change out convention here a bit. 

However, we will most probably have a 3rd group of Dock items in future, which may make this a good decision (?). The concept which is not fully designed in details is that user may pin some custom things into the dock. For example, they may pin a shortcut to an object or to a place (app, visualization, etc…) in the CRM. They may wanna do so to make their shortcut accessible from anywhere within the app, and remove the need of going back to for example the dashboard, to access a shortcut.

With this in mind, it may already be a good plan to divide the Dock items into 2 groups now, and add a 3rd group later when we are ready with the design. We know we will have something like this, but don't know how it will work in details yet.

Reply:
I'm a little divided on this myself. Mostly I just didn't like that you were supposed to set a certain property only on the first of the "special" items. If you'd set the property on all items that should go in the bottom, I'd be much more ok with it.

Reply:
Sum up of discussion from standup:

I was considering merging these two arrays, and adding a property that should be set on all items that should go in the footer. But during standup Kia showed some ideas for the future, where we'd have a kind of "history" of previous visited locations in the app, or maybe user-pinned locations, between the "head" and the "footer" items, and then I felt that having them split is probably quite a good idea. Merging the arrays would just mean we need to loop through them and pick out what to render where, every time we updated the array.